### PR TITLE
use a more modern golang with helm-docs

### PR DESCRIPTION
**Pull request by garypen**
Recent updates to helm-docs mean that the golang on the OS is too old.

Download a version from the golang website and use that.

fixes: #4025 

---

**Checklist**

Complete the checklist (and note appropriate exceptions) before the PR is marked ready-for-review.

- [x] Changes are compatible[^1]
- [ ] Documentation[^2] completed

**Exceptions**

*Note any exceptions here*

**Notes**

[^1]: It may be appropriate to bring upcoming changes to the attention of other (impacted) groups. Please endeavour to do this before seeking PR approval. The mechanism for doing this will vary considerably, so use your judgement as to how and when to do this.
[^2]: Configuration is an important part of many changes. Where applicable please try to document configuration examples.
[^3]: Tick whichever testing boxes are applicable. If you are adding Manual Tests, please document the manual testing (extensively) in the Exceptions.

### CODE REVIEW

**Review by BrynCooke — APPROVED** [review]

File: .changesets/maint_garypen_4025_helm_docs.md [hunk 1]
```diff
@@ -0,0 +1,9 @@
+### Use a more modern golang with helm-docs ([Issue #4025](https://github.com/apollographql/router/issues/4025))
+
+Recent updates to helm-docs mean that the golang on the Operating System is too old.
+
+Download a version from the golang website and use that.
+
+fixes: #4025 
+
+By [@garypen](https://github.com/garypen) in https://github.com/apollographql/router/pull/4027
```
Comment (BrynCooke) [inline .changesets/maint_garypen_4025_helm_docs.md:9]: Let's remove this changelog entry. It's not user actionable.

**Review by BrynCooke — CHANGES_REQUESTED** [review]

File: .changesets/maint_garypen_4025_helm_docs.md (hunk 1 above)
Comment (garypen) [reply .changesets/maint_garypen_4025_helm_docs.md:9]: Isn't that the case for most chores? I thought we had the maintenance section to let people know what we'd been doing.

Comment (BrynCooke) [reply .changesets/maint_garypen_4025_helm_docs.md:9]: At least it seems to be the trend right now to not include these.

Will add a suggestion if we're going to retain it.

File: .changesets/maint_garypen_4025_helm_docs.md [hunk 2]
```diff
@@ -0,0 +1,9 @@
+### Use a more modern golang with helm-docs ([Issue #4025](https://github.com/apollographql/router/issues/4025))
+
+Recent updates to helm-docs mean that the golang on the Operating System is too old.
+
+Download a version from the golang website and use that.
+
+fixes: #4025
```
Comment (BrynCooke) [inline .changesets/maint_garypen_4025_helm_docs.md:7]:
```suggestion
Fix CI builds failures due to outdated golang version no longer being supported by helm docs.

```

File: .changesets/maint_garypen_4025_helm_docs.md (hunk 1 above)
Comment (garypen) [reply .changesets/maint_garypen_4025_helm_docs.md]: I removed it.

File: .changesets/maint_garypen_4025_helm_docs.md (hunk 2 above)
Comment (BrynCooke) [reply .changesets/maint_garypen_4025_helm_docs.md]: Suggestion, `fixes: #4025` needs to go regardless.

**Review by BrynCooke — APPROVED** [review]

**Pull request merged**